metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1264,3 +1271,58 @@ def visit_semi_additive_join_node(self, node: SemiAdditiveJoinNode) -> SqlDataSe
                 order_bys=(),
             ),
         )
+
+    def visit_append_row_number_column_node(self, node: AppendRowNumberColumnNode) -> SqlDataSet:
+        """Appends a row number identifier column to the data set."""
+        input_data_set: SqlDataSet = node.parent_node.accept(self)
+        input_data_set_alias = self._next_unique_table_alias()
+
+        row_number_spec = ExtraSpec.from_name("mf_row_number")
+        row_number_spec_column_name = self._column_association_resolver.resolve_extra_spec(row_number_spec)
+
+        input_sql_column_references = input_data_set.instance_set.transform(
+            CreateSqlColumnReferencesForInstances(input_data_set_alias, self._column_association_resolver)
+        )
+        # Build enumerated column using ROW_NUMBER function
+        row_number_select_column = SqlSelectColumn(
+            expr=SqlWindowFunctionExpression(
+                sql_function=SqlWindowFunction.ROW_NUMBER,
+                order_by_args=[SqlWindowOrderByArg(expr=x) for x in input_sql_column_references],
```

Comment:
There will have to be a `FilterElementsNode` before this node so we don't have a long list of columns here.